src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -468,9 +468,16 @@ void SpatialPooler::initialize(
 }
 
 
-void SpatialPooler::compute(const SDR &input, const bool learn, SDR &active) {
+void SpatialPooler::compute(SDR &input, const bool learn, SDR &active) { //TODO make input const again
   input.reshape(  inputDimensions_ );
   active.reshape( columnDimensions_ );
+
+  //dropout, apply noise to input
+  input.addNoise2(0.01f, rng_); //TODO apply at synapse level in Conn?
+  //TODO fix for probability << input.size
+  //TODO apply killCells to active output? 
+  //TODO apply dropout to segments? (so all are: synapse, segment, cell/column)
```

Comment:
proof of concept dropout applied to input (as noise) and output (as killCells). 

- I'd prefer this be applied in Connections (in adaptSegment?) 
- where to apply?
  - ideally all of: SP, TM. & synapse, segment, cell, column
  - but that would be computationally infeasible, so..?